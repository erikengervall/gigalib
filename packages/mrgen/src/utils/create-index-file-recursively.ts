import fs from 'node:fs';
import path from 'node:path';

import { getAPIBit } from './get-api-bit';

// recursively create all index.ts files in dir
export function createIndexFileRecurse({
  dir,
  withAlias,
}: {
  dir: string;
  withAlias: boolean;
}) {
  console.debug(
    '[createIndexFileRecurse] Creating index file for directory',
    dir,
  );
  const filesAndFolders = fs.readdirSync(dir).filter((file) => {
    return (
      !['index.ts', '__snapshots__'].includes(file) &&
      !file.endsWith('.test.ts') &&
      !file.endsWith('.spec.ts')
    );
  });
  const indexFileContent = filesAndFolders
    .map((file) => {
      /**
       * > require('path').parse("a/b/c.ts")
       * { root: '', dir: 'a/b', base: 'c.ts', ext: '.ts', name: 'c' }
       */
      const { name } = path.parse(file);
      const alias = withAlias ? ` as ${getAPIBit(file)} ` : '';
      return `export * ${alias} from './${name}';`;
    })
    .join('\n');

  const indexFilePath = path.join(dir, 'index.ts');
  console.debug('[createIndexFileRecurse] Creating index file', indexFilePath);
  fs.writeFileSync(indexFilePath, indexFileContent, 'utf-8');

  filesAndFolders
    .filter((file) => fs.lstatSync(path.join(dir, file)).isDirectory())
    .forEach((file) => {
      createIndexFileRecurse({ dir: path.join(dir, file), withAlias });
    });
}
