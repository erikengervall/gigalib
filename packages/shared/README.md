# @engervall/shared

Utilities and stuff.
