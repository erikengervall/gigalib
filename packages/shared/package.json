{
  "name": "@engervall/shared",
  "version": "1.0.6-alpha.0",
  "main": "dist/index.js",
  "repository": "git@github.com:erikengervall/gigalib.git",
  "author": "Erik Engervall <erik.engervall@gmail.com>",
  "scripts": {
    "----- CI": "",
    "ci:build": "tsc",
    "ci:format": "prettier --check './src/**/*.{ts,tsx}' --ignore-path ../../.prettierignore",
    "ci:lint": "eslint 'src/**/*.{ts,tsx}'",
    "ci:test:unit": "jest",
    "----- LOCAL DEV": "",
    "dev": "rm -rf dist && tsc --watch",
    "----- MISC": "",
    "format:write": "prettier --write './src/**/*.{ts,tsx}'",
    "clean": "rm -rf dist",
    "reinstall": "yarn clean && rm -rf node_modules"
  },
  "dependencies": {
    "qs": "^6.11.2",
    "zod": "^3.21.4",
    "zod-validation-error": "^1.3.1"
  }
}
