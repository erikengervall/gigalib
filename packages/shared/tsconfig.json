{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "lib": ["es2021"],
    "module": "CommonJS",
    "noEmit": false,
    "sourceMap": true,
    "target": "ES6",

    "baseUrl": "src",
    "outDir": "dist",
    "rootDir": "src"
  },
  "include": ["src", "../../scripts/change-case.ts"],
  "exclude": [
    "_warmup",
    ".cache",
    ".next",
    ".serverless",
    ".vscode",
    ".vscode",
    ".webpack",
    "*.dist*",
    "*.spec*",
    "*.test*",
    "build",
    "coverage",
    "cypress-coverage",
    "dist",
    "jest-coverage",
    "node_modules"
  ]
}
