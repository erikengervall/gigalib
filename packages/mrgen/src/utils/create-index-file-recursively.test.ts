import { fsSnapshots } from '@engervall/shared';

import { createIndexFileRecurse } from './create-index-file-recursively';

it('should create index files recursively', () => {
  createIndexFileRecurse({ dir: __dirname, withAlias: true });

  fsSnapshots();
});
