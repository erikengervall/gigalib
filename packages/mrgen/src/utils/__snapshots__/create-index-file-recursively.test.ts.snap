// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create index files recursively: lstatSync 1`] = `
[
  [
    "/Users/noaa/dev/gigalib/packages/mrgen/src/utils/module-1.ts",
  ],
  [
    "/Users/noaa/dev/gigalib/packages/mrgen/src/utils/module-2.ts",
  ],
  [
    "/Users/noaa/dev/gigalib/packages/mrgen/src/utils/module-3.ts",
  ],
]
`;

exports[`should create index files recursively: mkdirSync 1`] = `[]`;

exports[`should create index files recursively: readdirSync 1`] = `
[
  [
    "/Users/noaa/dev/gigalib/packages/mrgen/src/utils",
  ],
]
`;

exports[`should create index files recursively: writeFileSync 1`] = `
[
  [
    "/Users/noaa/dev/gigalib/packages/mrgen/src/utils/index.ts",
    "export *  as Module_1  from './module-1';
export *  as Module_2  from './module-2';
export *  as Module_3  from './module-3';",
    "utf-8",
  ],
]
`;
